### PR TITLE
New module: directional-mean.

### PR DESCRIPTION
**Pull request by zhuang-hao-ming**
#1342 

As stated in the issue. This module is used to calculate **average direction** of a set of lines.
At the same time, it also calculates another number **circular variance** to represent the variation in
direction between the lines.
The last point, this module supports both projection and geographic coordinates.

## example

The result is a line(the red one) whose length is the average length of all lines and whose center is the 
centroid of  the feature set.

directional_mean

directional_mean1

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Have read [How To Contribute](https://github.com/Turfjs/turf/blob/master/CONTRIBUTING.md#how-to-contribute).
- [x] Run `npm test` at the sub modules where changes have occurred.
- [x] Run `npm run lint` to ensure code style at the turf module level.

Submitting a new TurfJS Module.

- [x] Overview description of proposed module.
- [x] Include JSDocs with a basic example.
- [x] Execute `./scripts/generate-readmes` to create `README.md`.
- [x] Add yourself to **contributors** in `package.json` using "Full Name <@GitHub Username>".

### CODE REVIEW

File: packages/turf-directional-mean/index.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
+"use strict";
```
Comment (DenisCarriere) [inline packages/turf-directional-mean/index.js]: Sounds like we would now need to add `index.js` in the global `.gitignore`

Comment (DenisCarriere) [reply packages/turf-directional-mean/index.js]: I'll merge this PR and delete it afterwards since I can't commit to @zhuang-hao-ming 's `master` branch

Comment (zhuang-hao-ming) [reply packages/turf-directional-mean/index.js]: Haha, I'm sorry, I forgot it. 
Adding it to the global `.gitignore` file seems to affect other modules that have not been refactored yet.

Comment (DenisCarriere) [reply packages/turf-directional-mean/index.js]: I think right now we've got far enough with Typescript that we can't really go back, so every "patch" should also involve refactoring the Typescript before fixing the module.

**Pull request merged**